libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -145,6 +144,35 @@ export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAd
         }
     }
 
+    /**
+     * Exposes the same functionality as [BotFrameworkAdapter.processActivityDirect](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/botFrameworkAdapter.ts).
+     *
+     * @param authorization Authorization in format: "Bearer [longString]".
+     * @param activity The activity to process.
+     * @param logic The logic function to apply.
+     * @returns a promise representing the asynchronous operation.
+     */
+    async processActivityDirect(
+        authorization: String,
+        activity: Activity,
+        logic: (context: TurnContext) => Promise<void>
+    ): Promise<void> {
+        const validatedActivity = validateAndFixActivity(ActivityT.parse(activity));
+
+        if (!validatedActivity.type) {
+            console.warn('BadRequest: Missing activity or activity type.');
+            throw new Error('Missing activity or activity type');
+        }
+
+        const authHeader = z.string().parse(authorization ?? '');
+
+        try {
+            await this.processActivity(authHeader, validatedActivity, logic);
```

Comment:
Shouldn't we return the invoke response?

Reply:
Regarding returning the invokeResponse, I'm having the following error. I think it's related to the end const that was created in the previous version. Anyway, in the deprecated class, only the middleware is called. In CloudAdapter, the processActivity method also does it.
<img width="1110" alt="image" src="https://user-images.githubusercontent.com/64815358/203810091-ff01404d-7614-48d8-b8f6-39be3e30c526.png">